### PR TITLE
ngctl: add check command and doc

### PR DESCRIPTION
**Pull request by kqzh**
- [x] add check command
- [x] support list command to list nebulacluster sub resources
- [x] add ngctl doc
- [x] add unit test

### CODE REVIEW

File: doc/user/ngctl_guide.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
+# Overview
+ngctl is a terminal cmd tool for nebula-operator, it has the following commands:
```
Comment (veezhang) [inline doc/user/ngctl_guide.md:3]: Is this better? `for nebula-operator` sounds like management `nebula-operator`.


`ngctl` is a terminal cmd tool for Nebula Graph managed by nebula-operator, it has the following commands:

File: doc/user/ngctl_guide.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+## ngctl use
+`ngctl use` specify a nebula cluster to use. By using a certain cluster, you may omit --nebulacluster option in many control commands.
+
+```
+Examples:
+  # specify a nebula cluster to use
+  ngctl use demo-cluster
+
+  # specify kubernetes context and namespace
```
Comment (veezhang) [inline doc/user/ngctl_guide.md:18]: I thanks this is just specify `namespace`, and don't support specify kubernetes context.

File: pkg/ngctl/cmd/check/check.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+		ngctl check pods --nebulacluster=nebula`)
+)
+
+const resourceType = "nebulacluster"
+
+type CheckOptions struct {
+	Namespace         string
+	NebulaClusterName string
+	ResourceType      string
+	AllNamespaces     bool
```
Comment (veezhang) [inline pkg/ngctl/cmd/check/check.go:54]: unused

File: pkg/ngctl/cmd/check/check.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+	checkExample = templates.Examples(`
+		# check whether the specified nebula cluster is ready
+		ngctl check 
+		
+		# check specified nebula cluster pods
+		ngctl check pods --nebulacluster=nebula`)
+)
+
+const resourceType = "nebulacluster"
```
Comment (veezhang) [inline pkg/ngctl/cmd/check/check.go:48]: It's also need in 'list.go', share it?

File: pkg/ngctl/cmd/check/check.go [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+// Validate validates the provided options
+func (o *CheckOptions) Validate(cmd *cobra.Command) error {
+	if o.NebulaClusterName == "" {
+		return cmdutil.UsageErrorf(cmd, "using 'ngctl use' or '--nebulacluster' to set nebula cluster first.")
+	}
+
+	if o.ResourceType == "" {
+		o.ResourceType = resourceType
+	}
+
```
Comment (veezhang) [inline pkg/ngctl/cmd/check/check.go:119]: It is need to check the `o.ResourceType` ?

File: pkg/ngctl/cmd/util/factory.go [hunk 6]
```diff
@@ -36,9 +36,9 @@ type (
 		ToRuntimeClient() (client.Client, error)
 		GetNebulaClusterNameAndNamespace(withUseConfig bool, args []string) (string, string, error)
 		GetNebulaClusterNamesAndNamespace(withUseConfig bool, args []string) ([]string, string, error)
-		// GetNebulaClusterName() (string, error)
+		GetNebulaClusterName() (string, error)
 		// GetNebulaClusterNameWithoutConfig() string
-		// GetNamespace() (string, error)
+		GetNamespace() (string, error)
```
Comment (veezhang) [inline pkg/ngctl/cmd/util/factory.go:41]: `GetNebulaClusterName` and `GetNamespace` may cause out of sync, can you use `GetNebulaClusterNameAndNamespace` instead?

File: pkg/ngctl/cmd/check/check.go (hunk 5 above)
Comment (kqzh) [reply pkg/ngctl/cmd/check/check.go:119]: If user don't specify the resourceType, we can use "nebulacluster" as a default type

**Comment by kqzh** [comment]
@veezhang Fixed all

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**